gluten-core/src/main/scala/io/glutenproject/expression/PredicateExpressionTransformer.scala

```diff
@@ -58,7 +58,9 @@ object InExpressionTransformer {
       values: Set[Any],
       valueType: DataType): ExpressionNode = {
     val expressionNodes = new java.util.ArrayList[ExpressionNode](
-      values
+      values.toSeq
+        // Sort elements for deterministic behaviours
+        .sortBy(value => if (value == null) "NULL"  else value.toString)
```

Comment:
It looks elements are sorted based the strings converted for values. Can we just do the sorting based on values? I'm assuming their type is comparable.

Reply:
Using `values.toSeq.sorted` directly seems to have compilation problems.

How about we follow the Spark logic in the following way:

```scala
      values.toSeq
        // Sort elements for deterministic behaviours
        .sortBy(Literal(_, valueType).toString())
```

Reply:
I think it may be ok. Thanks!